### PR TITLE
Tweak omnibus build

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

* Ignore gemfile lock for omnibus
* Make the omnibus build idempotent

### CODE REVIEW

File: omnibus/config/software/datadog-agent-finalize.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@
         move "#{install_dir}/etc/datadog-agent/datadog.yaml.example", "/etc/datadog-agent"
         move "#{install_dir}/etc/datadog-agent/trace-agent.conf", "/etc/datadog-agent/trace-agent.conf.example"
         move "#{install_dir}/etc/datadog-agent/process-agent.conf", "/etc/datadog-agent/process-agent.conf.example"
-        move "#{install_dir}/etc/datadog-agent/conf.d", "/etc/datadog-agent"
+        move "#{install_dir}/etc/datadog-agent/conf.d", "/etc/datadog-agent", :force=>true
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-finalize.rb]: in a scenario where you're running the omnibus build twice in the same environment, does this work properly when you add, remove or modify a file in `#{install_dir}/etc/datadog-agent/conf.d` between the 2 builds?

Just want to make the assumption of idempotency is correct :)

File: omnibus/Gemfile [hunk 2]
```diff
@@ -1,11 +1,8 @@
 source 'https://rubygems.org'
 
 # Install omnibus
-# TODO: allow to specify the branch as an env var
 gem 'omnibus', git: 'https://github.com/DataDog/omnibus-ruby.git', branch: 'datadog-5.5.0'
 
 # Use Chef's software definitions. It is recommended that you write your own
 # software definitions, but you can clone/fork Chef's to get you started.
-# TODO: allow to specify the branch as an env var
```
Comment (olivielpeau) [inline omnibus/Gemfile]: 🍰  thanks!

**Review by olivielpeau — COMMENTED** [review]
one comment on the idempotency (I'm still a bit cautious about saying that the build is idempotent ;) )

File: omnibus/config/software/datadog-agent-finalize.rb (hunk 1 above)
Comment (mfpierre) [reply omnibus/config/software/datadog-agent-finalize.rb]: You'll likely lose the changes you made in `{install_dir}/etc/datadog-agent/conf.d`

Maybe idempotency is a bit too strong, "make sure the build doesn't crash when run twice" is more accurate 😉

Comment (olivielpeau) [reply omnibus/config/software/datadog-agent-finalize.rb]: ok thanks for checking :) just want to make sure we don't expect omnibus to build "clean" packages on "unclean" environments

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**